libnodegl/backends/gl/gctx_gl.c

```
@@ -385,7 +389,15 @@ static int gl_init(struct gctx *s)
     int ret;
     const struct ngl_config *config = &s->config;
     struct gctx_gl *s_priv = (struct gctx_gl *)s;
-
+#ifdef ENABLE_CAPTURE
+    DEBUG_CAPTURE = (getenv("DEBUG_CAPTURE") != NULL);
+    if (DEBUG_CAPTURE) {
+        if (init_capture() != 0) {
+            LOG(ERROR, "could not initialize GPU capture");
+            DEBUG_CAPTURE = 0;
```

you could raise the error here

---

you mean return with an error code?

---

updated